src/shims/unix/fd.rs

```diff
@@ -463,4 +487,116 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         this.try_unwrap_io_result(result)
     }
+
+    fn pread(
+        &mut self,
+        fd: i32,
+        buf: Pointer,
+        count: u64,
+        offset: i128,
+    ) -> InterpResult<'tcx, i64> {
```

Comment:
This is largely a duplicate of `read`, right? Instead of duplicating all that code, would it be worth making this a single function? It could take an `Option` offset to indicate which mode to use.